src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -42,6 +42,28 @@ public Task<bool> IsEnabledAsync<TContext>(string feature, TContext appContext)
             return IsEnabledAsync(feature, appContext, true);
         }
 
+        public async Task<IEnumerable<string>> GetFeatureNamesAsync()
```

Comment:
Since this is all operating in memory, why does it need to be an async function?

Reply:
Alright, user can provide custom IFeatureSettingsProvider.

---
In reply to: [382884552](https://github.com/microsoft/FeatureManagement-Dotnet/pull/36#discussion_r382884552) [](ancestors = 382884552)

Reply:
Right the intent is to design around the capability for async feature settings provider. Although, currently `IFeatureSettingsProvider` is not public. That would be a different PR.